build_multi_arch_docker.sh

```diff
@@ -0,0 +1,97 @@
+#!/bin/bash
+# This script builds and deploys multi-architecture docker images.
+#
+# Can be run stand-alone from the command line like so:
+#   DOCKER_USER=... DOCKER_PASSWORD=... DOCKER_BASE=... DOCKER_PLATFORMS=... \
+#   TAGS=... \
+#   (source ./build_multi_arch_docker.sh; build-multi-arch-docker::main)
+# Where the environment variables are:
+#   DOCKER_USER: Docker Hub user name to which to push images.
+#   DOCKER_PASSWORD: Docker Hub password for above user.
+#   DOCKER_BASE: Docker image base name (user/image).
+#   TAGS: A set of image tags to which to deploy. Appended to DOCKER_BASE.
+#   DOCKER_PLATFORMS: Docker buildx platforms (see `docker buildx ls`).
+# Example:
+#   DOCKER_USER=foo DOCKER_PASSWORD=bar DOCKER_BASE=foo/shfmt TAGS=latest \
+#   (source ./build_multi_arch_docker.sh; build-multi-arch-docker::main)
+
+function build_multi_arch_docker::install_docker_buildx() {
```

Comment:
This command makes heavy use of sudo, and assumes a Debian-based distro. I think it belongs in the CI script directly, not here. We don't want to encourage users to modify their system as part of a "build" script, even less so when it's part of the recommended "main" command.

Reply:
Also, did you check if the Docker version in CI is too old? Because it might not be.

Reply:
It sounds like you'd prefer everything that is in build_multi_arch_docker.sh inside the ci.yml file? That's fine with me.

Reply:
Yeah. The only thing that's really directly useful to users is the "build" function, but that seems like a one-liner anyway.

Reply:
> Also, did you check if the Docker version in CI is too old? Because it might not be.

AFAICT the installed version of docker does not support buildx, even when experimental mode is turned on.
It is apparently from a different package, not docker-ce:
```
dpkg -S /usr/bin/docker
moby-cli: /usr/bin/docker
```
and that doesn't cut it:
```
export DOCKER_CLI_EXPERIMENTAL=enabled
docker buildx
docker: 'buildx' is not a docker command.
```

Reply:
Gotcha, thanks for checking